### ISSUE TITLE
Api routes - Could not find files for /api/gallery-data in .next/build-manifest.json {}

### ISSUE THREAD

**Issue by francisrod01** [zeit/next.js#9969]
# Bug report

## Describe the bug

Here's the prompt log as debug:

```bash
> travel-website@0.0.1 dev /home/paneladm/myproject
> next "src/app/"

[ wait ]  starting the development server ...
[ info ]  waiting on http://localhost:3000 ...

Warning, FIREBASE_CONFIG and GCLOUD_PROJECT environment variables are missing. Initializing firebase-admin will fail
Defining routes from exportPathMap
> Using "webpackDevMiddleware" config function defined in next.config.js.
> Using external babel configuration
> Location: "/home/paneladm/myproject/src/app/babel.config.js"
[ ready ] compiled successfully - ready on http://localhost:3000
[ wait ]  compiling ...
[ ready ] compiled successfully - ready on http://localhost:3000
[ event ] build page: /_error
[ wait ]  compiling ...
[ ready ] compiled successfully - ready on http://localhost:3000
[ event ] build page: /api/gallery-data
[ wait ]  compiling ...
[ ready ] compiled successfully - ready on http://localhost:3000
Could not find files for /api/gallery-data in .next/build-manifest.json
{}
TypeError: res.json is not a function
    at _default (/home/paneladm/myproject/dist/functions/next/server/static/development/pages/api/gallery-data.js:221:7)
    at processChild (/home/paneladm/myproject/node_modules/react-dom/cjs/react-dom-server.node.development.js:3204:14)
    at resolve (/home/paneladm/myproject/node_modules/react-dom/cjs/react-dom-server.node.development.js:3124:5)
    at ReactDOMServerRenderer.render (/home/paneladm/myproject/node_modules/react-dom/cjs/react-dom-server.node.development.js:3598:22)
    at ReactDOMServerRenderer.read (/home/paneladm/myproject/node_modules/react-dom/cjs/react-dom-server.node.development.js:3536:29)
    at renderToString (/home/paneladm/myproject/node_modules/react-dom/cjs/react-dom-server.node.development.js:4245:27)
    at render (/home/paneladm/myproject/node_modules/next/dist/next-server/server/render.js:82:16)
    at renderPage (/home/paneladm/myproject/node_modules/next/dist/next-server/server/render.js:332:20)
    at Object.ctx.renderPage (/home/paneladm/myproject/dist/functions/next/server/static/development/pages/_document.js:6100:20)
    at _callee2$ (/home/paneladm/myproject/dist/functions/next/server/static/development/pages/_document.js:4920:28)
    at tryCatch (/home/paneladm/myproject/node_modules/regenerator-runtime/runtime.js:45:40)
    at Generator.invoke [as _invoke] (/home/paneladm/myproject/node_modules/regenerator-runtime/runtime.js:271:22)
    at Generator.prototype.(anonymous function) [as next] (/home/paneladm/myproject/node_modules/regenerator-runtime/runtime.js:97:21)
    at asyncGeneratorStep (/home/paneladm/myproject/dist/functions/next/server/static/development/pages/_document.js:239:24)
    at _next (/home/paneladm/myproject/dist/functions/next/server/static/development/pages/_document.js:261:9)
    at /home/paneladm/myproject/dist/functions/next/server/static/development/pages/_document.js:268:7
```

## To Reproduce

Just create a file in `/pages/api/file.js` and try to call for it.

The `pages/api/gallery-data.js` is below:

```js
import galleryData from './gallery-data.json';

export default (req, res) => {
  res.json(galleryData);
}
```

I cleaned up my `next.config.js` file to better reproduce this bug, and I'm going to share it here:

```js
const { parsed: localEnv } = require('dotenv').config();
const webpack = require('webpack');
const withTM = require('next-transpile-modules');
const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');

module.exports = {
  distDir: '../../dist/functions/next',
  webpack: (config, options) => {
    // Alias @material-ui/core peer dependency imports form the following modules to our sources.
    config = withTM({
      transpileModules: ['notistack', '@material-ui/pickers', 'material-table'],
    }).webpack(config, options);

    const plugins = config.plugins.concat([
      new webpack.EnvironmentPlugin({
        'LIB_VERSION': JSON.stringify(pkg.version),
        ...useEnvs,
      }),
    ]);

    config.resolve.alias['react-dom$'] = 'react-dom/profiling';
    config.resolve.alias['scheduler/tracing'] = 'scheduler/tracing-profiling';

    // next includes node_modules in webpack externals. Some of those have dependencies
    // on the aliases defined above. If a module is an external those aliases won't be used.
    // We need tell webpack to not consider those packages as externals.
    if (options.isServer) {
      const [nextExternals, ...externals] = config.externals;

      if (externals.length > 0) {
        // currently not the case but other next plugins might introduce additional
        // rules for externals. We would need to handle those in the callback
        throw new Error('There are other externals in the webpack config.');
      }

      config.externals = [
        (context, request, callback) => {
          return nextExternals(context, request, callback);
        },
      ];
    }

    return Object.assign({}, config, {
      plugins,
      node: {
        fs: 'empty',
      },
      module: Object.assign({}, config.module, {
        rules: config.module.rules.concat([
          {
            test: /\.(css|md)$/,
            loader: 'emit-file-loader',
            options: {
              name: 'dist/[path][name].[ext]',
            },
          },
          {
            test: /\.(css|md)$/,
            loader: 'raw-loader',
          },
        ]),
      }),
    });
  },
  webpackDevMiddleware: config => config,
  exportTrailingSlash: true,

  // Next.js provides a `defaultPathMap` argument, we could simplify the logic.
  // However, we don't in order to prevent any regression in the `findPages()` method.
  exportPathMap: () => {
    const pages = findPages();
    const map = {};

    function traverse(pages2, userLanguage) {
      const prefix = userLanguage === 'en' ? '' : `/${userLanguage}`;

      pages2.forEach(page => {
          if (!page.children) {
            map[`${prefix}${page.pathname}`] = {
              page: page.pathname,
              query: {
                userLanguage,
              },
            };
            return;
          }

          traverse(page.children, userLanguage);
      });
    }

    return map;
  },
  onDemandEntries: {
    // Period (in ms) where the server will keep pages in the buffer
    maxInactiveAge: 120 * 1e3, // default 25s
    // Number of pages that should be kept simultaneously without being disposed
    pagesBufferLength: 3, // default 2
  },
}
```

## System information

- OS: Debian Stretch
- Browser Google Chrome Version 79.0.3945.88 (Official Build) (64-bit)
- Version of Nodejs: 8.16.2
- Version of Next.js: 9.1.7

**Comment by francisrod01** [comment]
Solved!
I'm using a custom path map to customize pages rendering (from Material-UI repo)

The solution is to ignore the `/api` pathname for pages index:

```js
exportPathMap: () => {
    const pages = findPages();
    const map = {};

    function traverse(pages2, userLanguage) {
      const prefix = userLanguage === 'en' ? '' : `/${userLanguage}`;

      pages2.forEach(page => {
        if (page.pathname.indexOf('/api') === -1) {
          if (!page.children) {
            ...
          }

          traverse(page.children, userLanguage);
        }
      });
    }

    return map;
  },
```

**Comment by JacksonMalloy** [comment]
For anyone else looking into this bug, I had a similar error message and solved it by removing an UNUSED IMPORT.

Keep in mind that if the code you're importing into one of the /pages is unused, it will be bundled client side and therefore cause errors when trying to load dependancies used with it. 

In my case, I was using postgres and it was erroring due to dns not being able to be resolved (which was a dependancy of the package). 

Hope that helps anybody else in the same situation!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
